Content.Shared/Gibbing/Systems/GibbingSystem.cs

```diff
@@ -0,0 +1,309 @@
+        float scatterImpulseVariance, Angle scatterCone, bool deleteTarget = true)
+    {
+        var localGibs = new List<EntityUid>();
+        var gibCount = 0;
+        var gibProtoCount = 0;
+        if (Resolve(target, ref gibable, logMissing: false))
+        {
+            gibCount = gibable.GibCount;
+            gibProtoCount = gibable.GibPrototypes.Count;
+        }
+        var gibAttemptEvent = new AttemptEntityGibEvent(target, gibCount, GibType.Drop);
+        RaiseLocalEvent(target, ref gibAttemptEvent);
+        switch (gibAttemptEvent.GibType)
+        {
+            case GibType.Skip:
+                return localGibs;
+            case GibType.Drop:
+                DropEntity(target, parentXform, randomSpreadMod, gibable, ref droppedEntities, flingEntity,
+                    scatterDirection, scatterImpulse, scatterImpulseVariance, scatterCone);
+                localGibs.Add(target);
+                return localGibs;
+        }
+
+        if (gibable != null && gibProtoCount > 0)
+        {
+            if (flingEntity)
+            {
+                for (var i = 0; i < gibAttemptEvent.GibletCount; i++)
+                {
+                    if (!TryCreateRandomGiblet(gibable, parentXform.Coordinates, false, out var giblet,
+                            randomSpreadMod))
+                        continue;
+                    FlingDroppedEntity(giblet.Value, scatterDirection, scatterImpulse, scatterImpulseVariance,
+                        scatterCone);
+                    droppedEntities.Add(giblet.Value);
+
+                }
+            }
+            else
+            {
+                for (var i = 0; i < gibAttemptEvent.GibletCount; i++)
+                {
+                    if (TryCreateRandomGiblet(gibable, parentXform.Coordinates, false, out var giblet, randomSpreadMod))
+                        droppedEntities.Add(giblet.Value);
+                }
+            }
+
+        }
+        _transformSystem.AttachToGridOrMap(target, Transform(target));
+        if (flingEntity)
+        {
+            FlingDroppedEntity(target, scatterDirection, scatterImpulse, scatterImpulseVariance, scatterCone);
+        }
+        var gibbedEvent = new EntityGibbedEvent(target, localGibs);
+        RaiseLocalEvent(target,ref gibbedEvent);
+        if (deleteTarget)
+            EntityManager.QueueDeleteEntity(target);
+        return localGibs;
+    }
+
+
+    public bool TryCreateRandomGiblet(EntityUid target, [NotNullWhen(true)] out EntityUid? gibletEntity ,
+        GibbableComponent? gibable = null, float randomSpreadModifier = 1.0f, bool playSound = true)
+    {
+        gibletEntity = null;
+        return Resolve(target, ref gibable) && TryCreateRandomGiblet(gibable, Transform(target).Coordinates,
+            playSound ,out gibletEntity, randomSpreadModifier);
+    }
+
+    public bool TryCreateAndFlingRandomGiblet(EntityUid target, [NotNullWhen(true)] out EntityUid? gibletEntity ,
```

Comment:
same for all these have `Entity<GibbableComponent?>` especially important with multiple entityuid params where confusion is much more likely